ext/js/lib/js.rb

```diff
@@ -132,6 +136,54 @@ def await
     promise = JS.global[:Promise].resolve(self)
     JS.promise_scheduler.await(promise)
   end
+
+  private
+
+  # Returns a JavaScript instance creation expression.
+  def to_construct(args)
+    "new #{constructor_name}(#{to_js_arguments_string(args)})"
```

Comment:
Could we use `Reflect.construct` instead of embedding arguments as string?

Reply:
`Reflect.construct` could be used. Thanks to this, it became much simpler.

Reply:
Much better now! Thanks :)